go/vt/servenv/grpc_codec.go

```diff
@@ -58,9 +63,11 @@ func (c *Codec) Marshal(v any) (mem.BufferSlice, error) {
 			return nil, err
 		}
 		return mem.BufferSlice{mem.NewBuffer(buf, defaultBufferPool)}, nil
+	case vtprotoMemBuffer:
+		return m.MarshalToMemBufferVT()
```

Comment:
This is not the right ordering; all the messages that implement `vtprotoMemBuffer` also implement `vtprotoMessage`, so this branch will never be used. You need to move this above.